File: backend/app/schemas/role_permissions.py
from pydantic import BaseModel, Field

from app.models.users import UserRole


class RolePermissionOptionRead(BaseModel):
    code: str
    label: str


class RolePermissionRead(BaseModel):
    role: UserRole
    permission_codes: list[str]


class RolePermissionMatrixRead(BaseModel):
    permissions: list[RolePermissionOptionRead]
    items: list[RolePermissionRead]


class RolePermissionUpdate(BaseModel):
    permission_codes: list[str] = Field(default_factory=list)

File: frontend/src/types/rolePermissions.ts
import type { UserRole } from '@/types/users';

export interface RolePermissionOptionRead {
  code: string;
  label: string;
}

export interface RolePermissionRead {
  permission_codes: string[];
  role: UserRole;
}

export interface RolePermissionMatrixRead {
  items: RolePermissionRead[];
  permissions: RolePermissionOptionRead[];
}

export interface RolePermissionUpdate {
  permission_codes: string[];
}
